icu4j/maven-migration/README_MAVEN.md

```diff
@@ -0,0 +1,26 @@
+# Maven progress
+
+- Packaging the many data files in a jar is slow (especially core)
+- The data files (.res & Co.) are spread out in the various modules.
+  So we would need a (non maven) step to distribute them.
+  Probably modify the step that produces the .jar files (in the C/C++ world)
+- The tests for localespi should become integration tests
+- We might want some integration tests to begin with
+- `LICENSE`, `security.policy`, anything else from `main/shared/`
```

Comment:
When I started on this the LICENSE was not a symlink, so I also didn't do it (not without checking).

And that was a bit of a problem, as the 3 artifacts that are published to Maven Central have LICENSE files in the .jar, which means that the LICENSE file should be in `<module>/src/main/resources`
That meant 3 copies to keep in sync.

But now that the repo already uses symlinks it means I can do the same for the module licenses.
Can do it in the script, or in step 2, when we merge the changes structure.

---

TBH, this file is more of a TODO than a README
(something that one would read to figure out what to do, how to run the migration scripts)
I'll update the content.
I also want to add some checks on the migration script, to require less (no) reading.
For example to check if it is running in the correct folder, and report if not.

Reply:
yeah either way. I went with the symlink because, other than installing the right git for windows, it meant less to do on the client side (no script to run) but once you're in ICU4J there's no reason you can't copy the LICENSE file out multiple times.   In fact, it's already referred to in multiple places.

Reply:
> Whoa, this is a lot more involved than I thought!

Some of the the more involved parts is eliminating some circular dependencies (`moveCircDepTestOutOfCore`).

For example some of the tests in `NumberFormatSpecificationTest.java` (in `core`) formats currencies.
But that formatting fails without `currdata`.
And `currdata` depends on `core`.

Ant goes around that by building all, then running the tests.
`core-test` depends on [ `core`, `currdata` ], and `currdata` depends on [ `core`]
So if the task order is build `core`, build `currdata`, build and run `core-tests`, all is good.

But maven tries to isolate the artifacts.
An artifact can't be deployed (and used) unless is passes all unit tests.
`core` and `core-test` are part of the same artifact (hence) the standard maven structure:
```
core
   src
      main / java
      main / resources
      tests / java
      tests / resources
```
So to build `currdata` the `core` has to be deployed, meaning `core-tests` pass. But that depends on `currdata`.
Hence the circular dependency.

That happens for many other tests (`ULocaleTest` depends on `langdata` & `regiondata`, etc)

This was mentioned in the "Cleanup of ICU4J Code" section of the "Build ICU4J with Maven" document.

---

Separating the code and data into `src/main/java` and `src/main/resources` is also a bit convoluted, but not much.

---

Let me know if you think I should document things a bit more.
Happy to do it.
Probably in a document, as this migration script is temporary.